ui/src/data/source.tsx

```diff
@@ -256,6 +360,53 @@ export class BackendSource implements Source {
       });
     });
   }
+
+  private makeEntityDataset(
+    requestedAttributes: string[],
+    occurrenceID: string,
+    cohort: Filter,
+    conceptSet: Filter | null
+  ): tanagra.EntityDataset {
+    const entity = findEntity(occurrenceID, this.config);
+    let cohortFilter = generateFilter(this, cohort, true);
+    if (!cohortFilter) {
+      throw new Error("Cohort filter is empty.");
+    }
+
+    if (occurrenceID) {
+      const primaryEntity = this.config.primaryEntity.entity;
+      cohortFilter = {
+        relationshipFilter: {
+          outerVariable: entity.entity,
+          newVariable: primaryEntity,
+          newEntity: primaryEntity,
+          filter: cohortFilter,
+        },
+      };
+    }
+
+    let filter = cohortFilter;
+    const conceptSetFilter = generateFilter(this, conceptSet, false);
+    if (conceptSetFilter) {
+      filter = {
+        arrayFilter: {
+          operator: tanagra.ArrayFilterOperator.And,
+          operands: [cohortFilter, conceptSetFilter],
+        },
+      };
+    }
+
+    return {
+      entityVariable: entity.entity,
+      selectedAttributes: requestedAttributes,
+      filter,
+      limit: 50,
+    };
+  }
+}
+
+function isInternalAttribute(attribute: string): boolean {
```

Comment:
Consider renaming this "isGeneratedAttribute" to be consistent with how we refer to these attributes in the API and indexing layers.

Reply:
Ah ok. Not this PR, but we should probably move that type of decoding logic to the backend also. I think the sql_function_wrapper and display value for attributes that I've added to the new underlay config version should help with some of this.